docs/src/design/stack/crypto_ops.md

```diff
@@ -0,0 +1,122 @@
+
+![rpperm](../../assets/design/stack/crypto_ops/RPPERM.png)
+
+In the above, $r$ (located in the helper register $h_0$) is the row address from the hash chiplet set by the prover nondeterministically.
+
+For the `RPPERM` operation, we define input and output values as follows:
+
+$$
+v_{input} = \alpha_0 + \alpha_1 \cdot op_{linhash} + \alpha_2 \cdot h_0 + \sum_{j=0}^{11} (\alpha_{j+4} \cdot s_{11-i})
+$$
+
+$$
+v_{output} = \alpha_0 + \alpha_1 \cdot op_{retstate} + \alpha_2 \cdot (h_0 + 7) + \sum_{j=0}^{11} (\alpha_{i+4} \cdot s_{11-i}')
+$$
+
+In the above, $op_{linhash}$ and $op_{retstate}$ are the unique [operation label](../chiplets/main.md#operation-labels) for initiating a linear hash and reading the full state of the hasher respectively. Also note that the term for $\alpha_3$ is missing from the above expressions because for Rescue Prime permutation computation the index column is expected to be set to $0$.
+
+Using the above values, we can describe constraint for the chiplet bus column as follows:
+
+>$$
+b_{chip}' \cdot v_{input} \cdot v_{output} = b_{chip} \text{ | degree } = 3
+$$
+
+The above constrain enforces that the specified input and output rows must be present in the trace of the hash chiplet, and that they must be exactly $7$ rows apart.
+
+The effect of this operation on the rest of the stack is:
+* **No change** starting from position $12$.
+
+## MPVERIFY
+The `MPVERIFY` operation verifies that a Merkle path from the specified node resolves to the specified root. This operation can be used to prove that the prover knows a path in the specified Merkle tree which starts with the specified node.
+
+Prior to the operation, the stack is expected to be arranged as follows (from the top):
+- Depth of the path, 1 element.
+- Index of the node, 1 element.
+- Value of the node, 4 elements.
+- Root of the tree, 4 elements.
+
+The Merkle path itself is expected to be provided by the prover non-deterministically (via the advice provider). If the prover is not able to provide the required path, the operation fails. Otherwise, the state of the stack does not change. The diagram below illustrates this graphically.
+
+![mpverify](../../assets/design/stack/crypto_ops/MPVERIFY.png)
+
+In the above, $r$ (located in the helper register $h_0$) is the row address from the hash chiplet set by the prover nondeterministically.
+
+For the `MPVERIFY` operation, we define input and output values as follows:
+
+$$
+v_{input} = \alpha_0 + \alpha_1 \cdot op_{mpver} + \alpha_2 \cdot r + \alpha_3 \cdot i + \sum_{j=0}^3 \alpha_{j+8} \cdot s_{6 - j}
+$$
+
+$$
+v_{output} = \alpha_0 + \alpha_1 \cdot op_{rethash} + \alpha_2 \cdot (h_0 + 8 \cdot s_0 - 1) + \sum_{j=0}^3\alpha_{j + 8} \cdot s_{10 - j}
+$$
```

Comment:
Shouldn't the stack index here be from s_9 down to s_6?

Reply:
Good catch! Fixed!